utils/utils.go

```diff
@@ -219,6 +220,38 @@ func CopyFileIfExists(source, dest string) error {
 	}})
 }
 
+func CopyToTempFile(sourceFileName, directory, pattern string) (string, error) {
+	destinationFile, err := os.CreateTemp(directory, pattern)
+	if err != nil {
+		return "", fmt.Errorf("failed to create temporary file: %w", err)
+	}
+	defer destinationFile.Close()
+	destinationFileName := destinationFile.Name()
+
+	sourceFile, err := os.Open(sourceFileName)
+	if err != nil {
+		return "", fmt.Errorf("failed to open %s: %w", sourceFileName, err)
+	}
+	defer sourceFile.Close()
+
+	if _, err = io.Copy(destinationFile, sourceFile); err != nil {
+		return "", fmt.Errorf("failed to copy %s to temporary file %s: %w", sourceFileName, destinationFileName, err)
+	}
+
+	// Preserve permissions
+	sourceFileInfo, err := os.Stat(sourceFileName)
```

Comment:
Do we need to care about attributes, etc, like running `cp --preserve=all`? I guess not in this specific case

Reply:
At the very least, we need to make the file executable

Since I was creating a more generic `CopyToTempFile` function, I thought that for the same price we could as well preserve the attributes

Reply:
That's what I was wondering. We're preserving the permissions. Just wondering if we'd need to preserve the extended attributes, for example. Certainly not needed here - maybe a function comment to note that the mode is preserved, but other file attributes are not? I'm not looking for the extended attributes to be implemented here, or attributes like owner/group/timestamp. Just a note in case anyone in the future decides to use this generic function

Reply:
Commentary added on top of the function :)